### ISSUE TITLE
random(3) and secure randomness

### ISSUE THREAD

**Issue by cyphar** [djpohly/libgfshare#2]
Really we should be using `getrandom(2)` or reading directly from `/dev/urandom` rather than using `random(3)` (which is deterministic). I don't really like this justification:

```c
static void
_gfshare_fill_rand_using_random( unsigned char* buffer,
                                 unsigned int count )
{
  unsigned int i;
  for( i = 0; i < count; ++i )
    buffer[i] = (random() & 0xff00) >> 8; /* apparently the bottom 8 aren't
                                           * very random but the middles ones
                                           * are
                                           */
}
```

I can prepare a patch for this if you like, but it's not clear to me whether this library is actually maintained.

**Comment by djpohly** [comment]
Yeah, I think that's the reason for the comment at [libgfshare.h:34-39](https://github.com/djpohly/libgfshare/blob/master/include/libgfshare.h#L34-L39).  It's already set up so you can assign a different function to `gfshare_fill_rand` if you want.

(I'm not the author of libgfshare, but I suspect the "not very random" remark is based on the "NOTES" section in the [Linux `rand`(3)](https://linux.die.net/man/3/rand) page.  Perhaps this code used to use that function instead.)
